sap-2-tier-marketplace-image/shared/os-disk-parts.json

```diff
@@ -100,7 +101,7 @@
             "uri": "[parameters('userImageVhdUri')]"
           },
           "vhd": {
-            "uri": "[concat('http://',parameters('userImageStorageAccount'),'.blob.core.windows.net/', resourceGroup().name, '/', parameters('vmName'), '-', variables('osDiskName'))]"
+            "uri": "[concat('http://',parameters('userImageStorageAccount'),'.blob.core.windows.net/', variables('resourceGroupLower'), '/', parameters('vmName'), '-', variables('osDiskName'))]"
```

Comment:
The diskname already contains a few components that are likely to be unique (e.g. SAP SID) so that should be sufficient.  If a container is really needed, use concat('vhds-', uniqueString(resourceGroup().id)) which will ensure naming requirements are met.